File: DI/DI-Dagger-Java/src/main/java/model/named/AppCollaborator.java
package model.named;

import javax.inject.Inject;
import java.util.UUID;

public class AppCollaborator implements IAppCollaborator {
    private final String id = UUID.randomUUID().toString();

    @Inject
    public AppCollaborator() {
    }

    @Override
    public String getName() {
        return "AppCollaborator id: " + id;
    }

    @Override
    public String toString() {
        return "AppCollaborator{" +
                "id='" + id + '\'' +
                '}';
    }
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/models/named/AppCollaborator.kt
package models.named

import java.util.*
import javax.inject.Inject

class AppCollaborator
@Inject constructor() : IAppCollaborator {
    private val id = UUID.randomUUID().toString()

    override fun getName(): String {
        return "AppCollaborator id: $id"
    }

    override fun toString(): String {
        return "AppCollaborator{" +
                "id='" + id + '\'' +
                '}'
    }
}
